packages/modal/src/Modal/Modal.tsx

```diff
@@ -25,13 +27,77 @@ import ModalView from './ModalView';
  * @param props.initialFocus By default, when a focus trap is activated the first element in the focus trap's tab order will receive focus. With this option you can specify a different element to receive that initial focus. Selector string (which will be passed to document.querySelector() to find the DOM node).
  * @param props.closeIconColor Choose between dark or light close icon. Default is dark.
  */
-const Modal = React.forwardRef((props: ModalProps, ref: ForwardedRef) => {
-  return (
-    <PopoverProvider>
-      <ModalView {...props} ref={ref} />
-    </PopoverProvider>
-  );
-});
+const Modal = React.forwardRef(
+  (
+    {
+      children,
+      closeIconColor,
+      contentClassName,
+      darkMode: darkModeProp,
+      open,
+      setOpen,
+      shouldClose = () => true,
+      size: sizeProp = ModalSize.Default,
+      ...rest
+    }: ModalProps,
+    forwardRef: React.Ref<HTMLDialogElement>,
+  ) => {
+    const { theme } = useDarkMode(darkModeProp);
+    const localRef = useRef<HTMLDialogElement>(null);
+    const mergedRef = useMergeRefs<HTMLDialogElement>([localRef, forwardRef]);
+
+    const consumerSetClose = () => {
+      if (shouldClose() && setOpen) {
+        setOpen(false);
+      }
+    };
+
+    const handleClose = () => {
+      if (shouldClose() && localRef.current != null) {
+        localRef.current?.close();
+
+        consumerSetClose();
+      }
+    };
+
+    useEffect(() => {
+      if (localRef.current != null) {
+        if (open) {
+          localRef.current?.showModal();
+        }
+
+        if (!open) {
+          handleClose();
+        }
+      }
+    }, [open, localRef, handleClose]);
+
+    const allowedSize = Object.values(ModalSize).includes(sizeProp);
+    const size = allowedSize ? sizeProp : ModalSize.Default;
+
+    // Escape key behavior handled by dialog element, out-of-the-box
+    // But to help keep consumer state in sync, we'll also listen for the escape key
+    useEscapeKey(consumerSetClose, { enabled: open });
+
+    return (
+      <PopoverProvider>
+        <LeafyGreenProvider>
+          <dialog
```

Comment:
wondering if we should deprecate contentClassName in favor of className since there's only one element now?